internal/data/testdata/trace_test.go

```diff
@@ -18,9 +18,61 @@ import (
 	"testing"
 
 	"github.com/open-telemetry/opentelemetry-collector/internal/data"
+	otlptrace "github.com/open-telemetry/opentelemetry-proto/gen/go/trace/v1"
 	"github.com/stretchr/testify/assert"
 )
 
+type traceTestCase struct {
+	name string
+	td   data.TraceData
+	otlp []*otlptrace.ResourceSpans
+}
+
+func generateAllTraceTestCases() []traceTestCase {
+	return []traceTestCase{
+		{
+			name: "no-resource-spans",
```

Comment:
nit: one can always grep so in practice it doesn't matter but all other names match function suffix, but not this one and "two-spans-same-resource" below.

Reply:
done.